redshift/redshift.go

```diff
@@ -30,17 +30,55 @@ type Redshift struct {
 	s3Info S3Info
 }
 
+// Table is our representation of a Redshift table
+// the main difference is an added metadata section and YAML unmarshalling guidance
+type Table struct {
+	Name    string    `yaml:"dest"`
+	Columns []ColInfo `yaml:"columns"`
+	Meta    Meta      `yaml:"meta"`
+}
+
+// Meta holds information that might be not in Redshift or annoying to access
+// in this case, we want to know the schema a table is part of
+// and the column which corresponds to the timestamp at which the data was gathered
+// NOTE: this will be useful for the s3-to-redshift worker, but is currently not very useful
+// same with the yaml info
+type Meta struct {
+	DataDateColumn string `yaml:"data_date_column"`
+	Schema         string `yaml:"schema"`
+}
+
+// ColInfo is a struct that contains information about a column in a Redshift database.
+// SortKey and DistKey only make sense for Redshift
+type ColInfo struct {
+	Ordinal    int    `yaml:"ordinal"`
+	Name       string `yaml:"dest"`
+	Type       string `yaml:"type"`
+	DefaultVal string `yaml:"defaultval"`
```

Comment:
Are we intentionally mixing snake case (data_date_column) above with ... nothing case here (defaultval)?